src/pudl/metadata/classes.py

```diff
@@ -204,7 +204,7 @@ def __repr_args__(self) -> list[tuple[str, Any]]:
 """Non-empty :class:`str` with no trailing or leading whitespace."""
 
 SnakeCase = pydantic.constr(
-    min_length=1, strict=True, regex=r"^[a-z][a-z0-9]*(_[a-z0-9]+)*$"
+    min_length=1, strict=True, regex=r"^[a-z_][a-z0-9_]*(_[a-z0-9]+)*$"
```

Comment:
in order to use the new double underscore name I stole this line from [this pr](https://github.com/catalyst-cooperative/pudl/pull/2818/files#)